### PR TITLE
Docs training

### PR DESCRIPTION
**Pull request by hflynn**
Transfers content from https://docs.google.com/document/d/1IE2x2-3-EHELb8YY1Rwgjb3geID2D7qkKQYJPZ_4AJE/edit#heading=h.lf6jehcj87xw to the contributing documentation.

### CODE REVIEW

**Comment by hflynn** [comment]
https://ci.openmicroscopy.org/view/Docs/job/CONTRIBUTING-merge-docs/1215/ updated with this content

File: contributing/editing-docs.rst [hunk 1]
```diff
@@ -0,0 +1,475 @@
+Editing the OME documentation
+=============================
+
+This guide assumes you are already familiar with :doc:`using-git` and GitHub
+and only covers where to find the sources, builds etc. you will need to edit
+and review any given content. Further information on the CI builds is
+available on the :doc:`ci-docs` page.
+
+Some of the live web documentation also features ‘Show on GitHub’ and ‘Edit on
+Github’ links in the lefthand menu which will take you directly to the source
```
Comment (jburel) [inline contributing/editing-docs.rst]: GitHub?

File: contributing/editing-docs.rst [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+file and allow you to edit within the GH interface and then open a PR (you
+should never use this functionality to edit autogenerated pages, see below for
+details of these).
+
+Overview
+--------
+
+This page covers technical documentation which is written in .rst files and
```
Comment (jburel) [inline contributing/editing-docs.rst]: quote around .rst?

File: contributing/editing-docs.rst [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+project, plus this set of ‘contributing developer’ documentation aimed at
+giving an overview of the OME process and workflows across products that might
+be of interest to external people, and the OME internal docs for
+internal-only private workflows and processes.
+
+Some of the content for these is either autogenerated or copied from external
+sources as described below. Formatting and style guidance can be found in the
+`README <https://github.com/openmicroscopy/ome-documentation/blob/develop/README.rst>`_
+for the ome-documentation repo, along with instructions for getting set up
+with sphinx.
```
Comment (jburel) [inline contributing/editing-docs.rst]: Sphinx

File: contributing/editing-docs.rst [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+    Product-specific information
+
+Bio-Formats documentation
+-------------------------
+
+Hosted at `<https://docs.openmicroscopy.org/bio-formats/>`_\{{version}} (plus
+latest redirect - :bf_doc:`docs.openmicroscopy.org/latest/bio-formats/ <>`)
+
+This documentation covers all aspects of Bio-Formats - using it with other
+tools, guidance of developers (specific to Bio-Formats rather than general OME
```
Comment (jburel) [inline contributing/editing-docs.rst]: maybe remove the bracket and write "specific guidance for B-F dev"

File: contributing/editing-docs.rst [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+
+The Sphinx documentation is in the ``/docs/sphinx/`` folder in the code
+repository at `<https://github.com/openmicroscopy/bioformats>`_.
+
+Content for autogenerating documentation is in ``/components/autogen/src/``.
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via ant. Within the Sphinx folder, ``make clean html``
```
Comment (jburel) [inline contributing/editing-docs.rst]: code quote around ant?

File: contributing/editing-docs.rst [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+^^^^^^^^^^^^^^^^^^^^^
+
+The following parts of the documentation are autogenerated from outside the
+docs folder:
+
+- Supported Formats table and format pages - generated from 
+  :bf_source:`format-pages.txt <components/autogen/src/format-pages.txt>` - 
+  **do not edit anything inside the** ``/formats`` **folder except the index**
+- Dataset structure table - generated from the readers
+- Metadata support table - generated from the reader
```
Comment (jburel) [inline contributing/editing-docs.rst]: readers?

File: contributing/editing-docs.rst [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+
+OME Contributing Developer documentation
+----------------------------------------
+
+Hosted at `<https://docs.openmicroscopy.org/contributing/>`_ (always latest).
+
+This covers the OME team processes and workflows that may be of interest to
+external contributors or other open source teams - information about what
+tools we use and how, rather than internal-only workflows (like standup prep)
+or anything which needs to be kept private which belongs in the internal docs
```
Comment (jburel) [inline contributing/editing-docs.rst]: I found the "which ... which" a bit heavy.

File: contributing/editing-docs.rst [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+^^^^^^
+
+The source files are at `<https://github.com/openmicroscopy/ome-documentation>`_, in the
+``/contributing`` folder.
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via ant. You can build locally using ``make clean html``
+as long as you have Sphinx and ant installed. There is further information on
```
Comment (jburel) [inline contributing/editing-docs.rst]: provided instead of as long as?

File: contributing/editing-docs.rst [hunk 9]
```diff
@@ -0,0 +1,475 @@
...
+
+Source
+^^^^^^
+
+`<https://github.com/openmicroscopy/ome-internal>`_ (private repository)
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via ant. You can build locally using ``make clean html``
```
Comment (jburel) [inline contributing/editing-docs.rst]: code marker for ant maybe

File: contributing/editing-docs.rst [hunk 10]
```diff
@@ -0,0 +1,475 @@
...
+^^^^^^
+
+The Sphinx documentation is in the ``/docs/sphinx/``
+folder in the code repository at
+`<https://github.com/ome/ome-cmake-superbuild>`_.
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via cmake. You can build locally using ``make clean
```
Comment (jburel) [inline contributing/editing-docs.rst]: code marker for cmake maybe

File: contributing/editing-docs.rst [hunk 11]
```diff
@@ -0,0 +1,475 @@
...
+Source
+^^^^^^
+
+The Sphinx documentation is in the ``/docs/sphinx/`` folder in the code
+repository at `<https://github.com/ome/ome-files-cpp>`_.
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via cmake. You can build locally using ``make clean
```
Comment (jburel) [inline contributing/editing-docs.rst]: code marker for cmake maybe

File: contributing/editing-docs.rst [hunk 12]
```diff
@@ -0,0 +1,475 @@
...
+Source
+^^^^^^
+
+`<https://github.com/openmicroscopy/ome-internal>`_ (private repository)
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via ant. You can build locally using ``make clean html``
+as long as you have Sphinx and ant installed.
```
Comment (jburel) [inline contributing/editing-docs.rst]: provided?

File: contributing/editing-docs.rst [hunk 13]
```diff
@@ -0,0 +1,475 @@
...
+Source
+^^^^^^
+
+All the source files are in the ``/omero/`` folder at
+`<https://github.com/openmicroscopy/ome-documentation>`_.
+
+Building locally
+^^^^^^^^^^^^^^^^
+
+The build uses Sphinx via ant. You can build locally using ``make clean html``
```
Comment (jburel) [inline contributing/editing-docs.rst]: code marker for ant?

File: contributing/editing-docs.rst [hunk 14]
```diff
@@ -0,0 +1,475 @@
...
+as long as you have Sphinx and ant installed. There is further information on
```
Comment (jburel) [inline contributing/editing-docs.rst]: provided?

File: contributing/editing-docs.rst [hunk 15]
```diff
@@ -0,0 +1,475 @@
...
+    -  Open a doc PR
+    -  Copy the generated walkthrough(s) under
+       `omero/sysadmins/unix/walkthrough <https://github.com/openmicroscopy/ome-documentation/tree/develop/omero/sysadmins/unix/walkthrough>`_
+    -  Adjust if required the start/end of the `literalinclude`
+
+OMERO.web installation separately from  OMERO.server:
+
+-  The walkthroughs are generated using ansible. The
+   `omeroweb-install README <https://github.com/ome/omeroweb-install/blob/master/README.md>`_
+   file contains instructions on how to generate the walkthrough
```
Comment (jburel) [inline contributing/editing-docs.rst]: walkthroughs

**Comment by joshmoore** [comment]
For those searching, https://ci.openmicroscopy.org/job/CONTRIBUTING-merge-docs/ws/src/contributing/_build/html/editing-docs.html which looks like a nice addition. Thanks, @hflynn

**Comment by hflynn** [comment]
Is that a good to merge @joshmoore ?

**Comment by joshmoore** [comment]
I didn't review the latest changes. Just a general :tada: I'd defer to @jburel

**Review by sbesson — APPROVED** [review]
Also think this is in a mergeable and deployable state. Thanks for making the effort to add this knowledge into user-facing documentation.

**Comment by hflynn** [comment]
JM is quite busy enough at the moment and I fixed all his nitpicks so I'll get this merged and deployed

**Comment by jburel** [comment]
Thanks @hflynn we can have a follow up if we find other problems

**Comment by hflynn** [comment]
I just found one - https://ci.openmicroscopy.org/view/Docs/job/CONTRIBUTING-latest-docs/543/warnings3Result/ I need to add ome-internal to the ignore list in the config

**Pull request merged**